src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -336,37 +343,40 @@ def sync_deploy_root_with_stage(
             "Performing a diff between the Snowflake stage and your local deploy_root ('%s') directory."
             % self.deploy_root
         )
-        diff: DiffResult = stage_diff(self.deploy_root, self.stage_fqn)
+        diff: DiffResult = compute_stage_diff(self.deploy_root, self.stage_fqn)
 
         files_not_removed = []
-        if len(paths_to_sync) > 0:
+        if local_paths_to_sync is not None:
```

Comment:
I agree that we shouldn't be sending in an empty array for `local_paths_to_sync` (None is better), but what happens if we do? Sensible error message? Does everything work the same as `None`?

Reply:
This change was really about removing the mutable default object, because Python is weird that way and it's bad practice. I ended up going with "if you pass a list, we use it" for the semantics. I can change things so that empty is the same as None, but right now you'd likely end up with an empty diff instead.

Reply:
OK, as long as it's clear in the docstrings I'm fine with that!

Reply:
Added full doc string, but since passing an empty list makes no sense, I flipped to the logic to make `None` equivalent to `[]`, it's a fair callout.